core/api/views.py

```
@@ -20,30 +20,37 @@
 
 
-class ProfileView(APIView):
-    """
-    Get user profile data.
-    """
-    permission_classes = [IsAuthenticated]
-    def get(self, request):
-        user = request.user
-        serializer = UserSerializer(user)   
+# Helper methods to reduce code duplication
+def create_success_response(message, data=None, status_code=status.HTTP_200_OK):
```

Ruff (0.8.2)

Missing return type annotation for public function `create_success_response`

(ANN201)